### ISSUE TITLE
Only first upstream server is used in forward plugin

### ISSUE THREAD

**Issue by martinhaus** [coredns/coredns#3900]
**What happened**:

The policy for selecting upstream servers in the forward plugin is not working properly. Only the first upstream server is always selected no matter which policy is set. The issue seems to occur in 1.6.7 and later.

The issue is introduced with this commit https://github.com/coredns/coredns/commit/5e04c272382a7d7aca15f8ac9a8100a373ca1d1e#diff-167a31fa4e59b31f81c85cd09c92415

I assume the issue is caused by using the `List` method  `policy.go` incorrectly.

**What you expected to happen**:

Requests should be made to multiple upstream servers based on the policy which is configured.
When looking at produced metrics I expect to see multiple upstream servers.

```
coredns_forward_requests_total{to="1.1.1.1:53"} 3
coredns_forward_requests_total{to="8.8.4.4:53"} 3
coredns_forward_requests_total{to="8.8.8.8:53"} 2

```

**How to reproduce it (as minimally and precisely as possible)**:

1. Configure multiple upstream servers

```
forward . 1.1.1.1 8.8.8.8 9.9.9.9 8.8.4.4 {
      force_tcp
      policy random
      #max_fails 0
      health_check 5s
  }
```

2. Make multiple requests
```
➜  ~ dig 2.mnt.sk -p 5300 @127.0.0.1 +tcp; dig 1.mnt.sk -p 5300 @127.0.0.1 +tcp; dig 0.mnt.sk -p 5300 @127.0.0.1 +tcp; dig 4.mnt.sk -p 5300 @127.0.0.1 +tcp; dig 5.mnt.sk -p 5300 @127.0.0.1 +tcp; dig 6.mnt.sk -p 5300 @127.0.0.1 +tcp; dig 7.mnt.sk -p 5300 @127.0.0.1 +tcp;

```

3. Looking at produced metrics we can see only one upstream server is used for all of them
```
➜  ~ curl -v localhost:9153/metrics| grep coredns_forward_requests_total

coredns_forward_requests_total{to="1.1.1.1:53"} 8

```

**Anything else we need to know?**:

**Environment**:

- the version of CoreDNS: 1.6.9
- Corefile:
```
.:5300 {
  whoami
  ready :8091
  health :8080
  prometheus :9153
  loadbalance round_robin
  loop
  log . {
    class all
  }
  errors
  forward . 1.1.1.1 8.8.8.8 9.9.9.9 8.8.4.4 {
      force_tcp
      policy random
      #max_fails 0
      health_check 5s
  }

```
- logs, if applicable:
- OS (e.g: `cat /etc/os-release`):
- Others:

**Comment by martinhaus** [comment]
I've created a PR for this.
https://github.com/coredns/coredns/pull/3901

I don't have that much Go experience and assume the change can be made prettier. Let me know if you have any tips or improvements.

**Comment by ghost** [comment]
+1

Tested it on `v1.6.6` and the same issue.

**Comment by martinhaus** [comment]
@Prajwal-Koirala `v1.6.6` works fine for us.
At least based on the prometheus metric `coredns_forward_request_count_total`.
We can see the requests are distributed between all specified upstreams

**Comment by chrisohaver** [comment]
Fixed in #3919
